### ISSUE TITLE
unfulfilled_lint_expectations in combination with proc macro `[wasm_bingen]`

### ISSUE THREAD

**Issue by threema-theo** [rust-lang/rust-clippy#14008]
### Summary

clippy throws false positives when using a proc macro (I only tested with `[wasm_bindgen]`.

I further observed the following:
- Removing `#[wasm_bindgen]` removes the false positive
- Removing `#[expect(...)]` correctly throws warnings
- Replacing `#[expect(...)]` with `#[allow(...)]` correctly removes the warnings.

I'm not 100%  sure whether it is a problem of clippy or wasm-bindgen. Please tell me if the latter should be the case.

### Lint Name

unfulfilled_lint_expectations

### Reproducer

I tried [this code](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=c5977b3a742ba778e0024dfa3576922b):

```rust
#![warn(clippy::missing_panics_doc)]
#![warn(clippy::missing_errors_doc)]


use wasm_bindgen::prelude::wasm_bindgen;

#[expect(clippy::missing_panics_doc, reason = "mock example")]
#[wasm_bindgen]
pub fn panic() {
    panic!("Here we go");
}

#[expect(clippy::missing_errors_doc, reason = "mock example")]
#[wasm_bindgen]
pub fn error() -> Result<usize, usize> {
    Err(1)
}
```

I saw this happen:

```
    Checking playground v0.0.1 (/playground)
warning: this lint expectation is unfulfilled
 --> src/main.rs:7:10
  |
7 | #[expect(clippy::missing_panics_doc, reason = "mock example")]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: mock example
  = note: `#[warn(unfulfilled_lint_expectations)]` on by default

warning: this lint expectation is unfulfilled
  --> src/main.rs:13:10
   |
13 | #[expect(clippy::missing_errors_doc, reason = "mock example")]
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: mock example

warning: `playground` (bin "playground") generated 2 warnings
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.05s
```

I expected to see this happen: Clippy throws no warnings.

When I remove the `expect` macros, Clippy correctly throws the warnings.

### Version

```text
1.86.0-nightly
2025-01-15 419b3e2d3e350822550e

(see [playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=c5977b3a742ba778e0024dfa3576922b)
```

**Comment by samueltardieu** [comment]
`wasm_bindgen` applies identical outer attributes to the autogenerated code, which will not fullfill the expectations. I have proposed a patch to `wasm_bindgen` which transforms those attribute into equivalent `#[allow]` attributes in <https://github.com/rustwasm/wasm-bindgen/pull/4409>.

**Comment by threema-theo** [comment]
Resolved in https://github.com/wasm-bindgen/wasm-bindgen/pull/4409 🥳 

Thanks a lot, @samueltardieu !
